models/multitask_centernet/multitask_centernet.py

```diff
@@ -0,0 +1,205 @@
+        for i in person_indices:
+            if person_confidences[i] > config['conf_thres_kp_person']:
+                pose = outs[person_rowinds[i], self.last_ind:].reshape((-1, 2))
+                pose[:, 0] = (pose[:, 0] - padw) * ratiow
+                pose[:, 1] = (pose[:, 1] - padh) * ratioh
+                poses.append(pose)
+        nd = len(poses)
+        poses = np.array(poses)
+        poses = np.concatenate((poses, np.zeros((nd, poses.shape[1], 1))), axis=-1)
+        for j in kp_indices:
+            box = kp_boxes[j]
+            x = box[0] + 0.5 * box[2]
+            y = box[1] + 0.5 * box[3]
+            pt_id = kp_classIds[j] - 1
+            pose_kps = poses[:, pt_id, :]
+            dist = np.linalg.norm(pose_kps[:, :2] - np.array([[x, y]]), axis=-1)
+            kp_match = np.argmin(dist)
+            if kp_confidences[j] > pose_kps[kp_match, 2] and dist[kp_match] < config['overwrite_tol']:
+                poses[kp_match, pt_id, :] = np.array([x, y, kp_confidences[j]])
+
+        for i in person_indices:
+            box = person_boxes[i]
+            left = box[0]
+            top = box[1]
+            width = box[2]
+            height = box[3]
+            frame = self.drawPred(frame, person_classIds[i], person_confidences[i], left, top, left + width,
+                                  top + height)
+
+        for pose in poses:
+            for seg in self.lines.values():
+                pt1 = (int(pose[seg[0], 0]), int(pose[seg[0], 1]))
+                pt2 = (int(pose[seg[1], 0]), int(pose[seg[1], 1]))
+                cv2.line(frame, pt1, pt2, (255, 0, 255), 1)
+            for x, y, c in pose:
+                if c > 0:
+                    cv2.circle(frame, (int(x), int(y)), 1, (0, 0, 255), 1)
+
+            #for x, y, c in pose[self.kp_face]:
+                #cv2.circle(frame, (int(x), int(y)), 1, (255, 0, 255), 1)
+        # for i in kp_indices:
+        #     box = kp_boxes[i]
+        #     left = box[0]
+        #     top = box[1]
+        #     width = box[2]
+        #     height = box[3]
+        #     frame = self.drawPred(frame, kp_classIds[i], kp_confidences[i], left, top, left + width, top + height)
+        return frame
+
+    def drawPred(self, frame, classId, conf, left, top, right, bottom):
+        # Draw a bounding box.
+        cv2.rectangle(frame, (left, top), (right, bottom), (0, 0, 255), thickness=1)
+
+        label = '%.2f' % conf
+        label = '%s:%s' % (self.classes[classId], label)
+
+        # Display the label at the top of the bounding box
+        labelSize, baseLine = cv2.getTextSize(label, cv2.FONT_HERSHEY_SIMPLEX, 0.5, 1)
+        top = max(top, labelSize[1])
+        # cv.rectangle(frame, (left, top - round(1.5 * labelSize[1])), (left + round(1.5 * labelSize[0]), top + baseLine), (255,255,255), cv.FILLED)
+        cv2.putText(frame, label, (left, top - 10), cv2.FONT_HERSHEY_SIMPLEX, 0.5, (0, 255, 0), thickness=1)
+        return frame
+
+    def detect(self, srcimg):

```

Comment:
Rename this function to `infer` and keep it simple like the following:
```python
def infer(self, image):
    input_blob = self.preprocess(image)

    self.model.setInput(input_blob)
    output_blob = self.model.forward(self.model.getUnconnectedOutLayersNames())

    results = self.postprocess(output_blob)
    return results
```

Here is [another example](https://github.com/opencv/opencv_zoo/blob/ae1d754a3ea14e4244fbea7d781cca2e18584035/models/human_segmentation_pphumanseg/pphumanseg.py#L44-L58) for reference.